### PR TITLE
[Flow] Add github exporter

### LINKED ISSUE #2316: Flow: Add github_exporter
Create `prometheus.integration.github_exporter` component

### PR DESCRIPTION
**Pull request by jcreixell**
#### PR Description

  - Adds a github exporter to the list of supported flow integrations.

#### Which issue(s) this PR fixes

Fixes #2316 

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.integration.github.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+---
+
+# prometheus.integration.github
+The `prometheus.integration.github` component embeds
+[github_exporter](https://github.com/infinityworks/github-exporter) for collecting statistics from GitHub.
+
+## Usage
+
+```river
+prometheus.integration.github "LABEL"{
```
Comment (spartan0x117) [inline docs/sources/flow/reference/components/prometheus.integration.github.md:19]:
```suggestion
prometheus.integration.github "LABEL" {
```

Comment (spartan0x117) [reply docs/sources/flow/reference/components/prometheus.integration.github.md:19]: Looks like I was just behind rfratto, ignore this :smile:

File: docs/sources/flow/reference/components/prometheus.integration.github.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+The following arguments can be used to configure the exporter's behavior.
+All arguments are optional. Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_url`    | `string` | The full URI of the github API. | `https://api.github.com` | no
+`repositories` | `list(string)` | A list of github repositories for which to collect metrics. | | no
+`organizations` | `list(string)` | A list of github organizations for which to collect metrics. | | no
+`users` | `list(string)` | A list of github users for which to collect metrics. | | no
+`api_token`    | `string` | A github authentication token that allows the API to be queried more often. Optional, but recommended. | | no
```
[the remark this answers is not in the document]
Comment (jcreixell) [reply docs/sources/flow/reference/components/prometheus.integration.github.md:33]: good catch

**Comment by jcreixell** [comment]
@rfratto thanks for the review, feel free to merge if you are happy with the changes

File: docs/sources/flow/reference/components/prometheus.integration.github.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+`api_url`    | `string` | The full URI of the GitHub API. | `https://api.github.com` | no
+`repositories` | `list(string)` | GitHub repositories for which to collect metrics. | | no
+`organizations` | `list(string)` | GitHub organizations for which to collect metrics. | | no
+`users` | `list(string)` | A list of GitHub users for which to collect metrics. | | no
+`api_token`    | `secret` | API token to use to authenticate against GitHub. | | no
+`api_token_file`    | `string` | File containing API token to use to authenticate against GitHub. | | no
+
+GitHub uses an aggressive rate limit for unauthenticated requests based on IP address. To allow more API requests, it is recommended to configure either api_token or api_token_file to authenticate against GitHub.
+
+When provided, api_token_file takes precedence over api_token.
```
Comment (rfratto) [inline docs/sources/flow/reference/components/prometheus.integration.github.md:38]:
```suggestion
GitHub uses an aggressive rate limit for unauthenticated requests based on IP address. To allow more API requests, it is recommended to configure either `api_token` or `api_token_file` to authenticate against GitHub.

When provided, `api_token_file` takes precedence over `api_token`.
```

**Pull request merged**